### DISCUSSION TITLE
Need some testers with two or more BMS

### DISCUSSION THREAD

**Discussion by mr-manuel** [Louisvdw/dbus-serialbattery#979] [General]
Hello,

for a code change I would need at least one tester which has connected two or more BMS via serial connection and where I can connect directly via SSH or TeamViewer. The test takes about one hour.

Who will be available? Please join the Discord channel for faster communication: https://discord.gg/YXzFB8rSgx

**Comment by coca7** [comment]
I have 3 JBD BMS's connected.  I am available today from 11:00am central standard till about 2:00 pm Standard.

**Reply by mr-manuel** [reply]
Can you write me on Discord? You find the link in my profile

**Reply by coca7** [reply]
I'm not super familiar with discord, but I do have a login.  I just used your link, send a friend request and messaged you.  User name = Corvus#4406

**Comment by seamaster101** [comment]
@mr-manuel  I have 4 X JKBMS connected via RS485 and also via BLE. How can i be of service?

**Reply by mr-manuel** [reply]
Can you write me on Discord?

**Reply by seamaster101** [reply]
I don't remember ever using Discord, but apparently I have an account…
I sent friend request to mrmanuel and drmanuel account. I believe if that is you, you should see the friend request from seamster 
If that is not the proper way to do it, please let me know how to contact you. 
I’m available for testing all they long today if that works for you. Just let me kno

**Reply by seamaster101** [reply]
Best would be to you to contact me the discord account display name is SeaMaster

**Comment by seamaster101** [comment]
Mr-Manauel,
I don't remember ever using Discord, but apparently I have an account…
I sent friend request to mrmanual account. I believe if that is you, you should see the friend request from seamster 
If that is not the proper way to do it, please let me know how to contact you. 
I’m available for testing all they long today if that works for you. Just let me know
Jordan

**Reply by mr-manuel** [reply]
I added you else you find a link to the Discord Server in my profile.

**Comment by ramack** [comment]
I am not on discord, but for sure willing to help. Right now I am running an older version, patched to poll slower than 1 second, as my 5 BMSs overload by VenusOS otherwise. So maybe the best it if you tell me what should be tested, then I can do it. Should also be easy to connect multiple BMSes to the same RS485, as my Heltec/YYBMS systems run already on individual addresses.

**Reply by mr-manuel** [reply]
Could you test `v1.2.20240227dev`? Option 6 with the latest installer script:

```bash
wget -O /tmp/install.sh https://raw.githubusercontent.com/Louisvdw/dbus-serialbattery/master/etc/dbus-serialbattery/install.sh

bash /tmp/install.sh
```

What to test:
* Is the driver starting correctly, if all BMS have an `unique identifier`?
* Is the driver NOT starting, if two or more BMS have the same `unique_identifier`?

**Reply by ramack** [reply]
I tried with `3) beta build "v1.2.20240227beta" (louisvdw's repo)` on my system with 5 Heltec/YYBMS, but no battery was detected, even after a manual reboot, but as expected that was a load issue and patching heltecmodbus.py to set `self.poll_interval = 5000` in `__init__` did the trick such that all 5 BMSs (which have a different serial number = unique_identifier) are detected and operating properly. Maybe we could add a setting to overwrite the poll_interval proposed by the bms driver?

Still the ESS did not work properly, as I use dbus-aggregate-batteries and this was deselected as controlling BMS in DVCC (and shown as unavailable) is uses DeviceInstance = 0, while the 5 serial BMSs have 1 to 5. Restarting dbus-aggregate-batteries brought back the "AggregateBatteries" value in the contolling BMS to make everything work. - Even after the next reboot, so I don't catch why it was needed at first. The discovery of the serial BMSs is a bit slow but who cares? - it should not happen that often.

Now checking the VRM IDs: They got "new random" VRM IDs (which is not surprising) so I adjusted the values in 
Settings/Devices/serialbattery_XXXXXXXXXXXXX/ClassAndVrmInstance using dbus-spy to make the VRMInstance match the labels on the batteries and reached what I was hoping for.

Last try was to swap two USB ports and also that works like a charm and the batteries keep their VRM ID. **So thanks a lot, this is a very helpful improvement!**

For the records: I am on VenusOS 3.13

**Reply by snow78** [reply]
i also use dbus-aggregate-batteries. i increased theSEARCH_TRIALS = 20                                      # Trials to identify of all batteries before exit and restart
READ_TRIALS = 20                                        # Trials to get consistent data of all batteries before exit and restart

but still this only helps when the serial driver take a bit longer to detect. still sometimes the aggregate batteries hangs when dbus-serial restarts. i assume that happens when Aggregate batteries reads the serial battery when it serial batteries fails at the first start. that is what i observed that often the serial batteries gets some read errors from the serial at first start and then restarts.

**Comment by mr-manuel** [comment]
I published a new beta release, which is also a release candidate. Please install as much people as possible :-)

https://github.com/mr-manuel/venus-os_dbus-serialbattery/releases/tag/v1.2.20240227beta

Could you test `v1.2.20240227dev`? Option 6 with the latest installer script:

```bash
wget -O /tmp/install.sh https://raw.githubusercontent.com/Louisvdw/dbus-serialbattery/master/etc/dbus-serialbattery/install.sh

bash /tmp/install.sh
```

Then select option 4.

**Comment by seamaster101** [comment]
I installed and tested this version today. All 4 BMSs have connected well and work. I tried with BLE connection as well. That so don’t work well. They connect but don’t update parameters. On Feb 27, 2024, at 01:08, Manuel ***@***.***> wrote:﻿
Could you test v1.2.20240227dev? Option 6 with the latest installer script:
wget -O /tmp/install.sh https://raw.githubusercontent.com/Louisvdw/dbus-serialbattery/master/etc/dbus-serialbattery/install.sh

bash /tmp/install.sh
What to test:

Is the driver starting correctly, if all BMS have an unique identifier?
Is the driver NOT starting, if two or more BMS have the same unique_identifier?

**Reply by AncoDiara** [reply]
Indeed I'm running several installations with 1 JKbms on bluetooth really well. but I cannot get Two JKbms'es working stable via bluetooth. I tried several builds and now running 1.2.20240227dev. But I need to reboot the cerbo-s gx every few hours to get updated info from JKbms. Already after few minutes one of the two JKbms'es becomes very slow in updating voltage and current info. Finally they stop both and after few  minutes or hours sometimes come back with good info. See yesterday charging SOC profile for example.
![Screenshot from 2024-03-15 23-43-10](https://github.com/Louisvdw/dbus-serialbattery/assets/120603200/233844f2-7df7-4afd-b4ba-155bb8657e0d)

**Comment by snow78** [comment]
Hi, still testers needed? i have 3*JKBMS with aggregate batteries and even the 1.0.20240102beta installed already (was looking for the stable name feature) this works. i have no issues with it

**Reply by mr-manuel** [reply]
Yes, just released also the beta version in louisvdw repo.

**Reply by snow78** [reply]
installed and works well. 3 BMS connected via Serial. custom names properly remembered. cant see so far any issues

**Comment by Approved-OK** [comment]
Testet the new driver.[v1.2.20240227 , CX OSV3.22
2 Daly BMS, seriell connection, BatteryAggregator.
problem is, that unique identifyer is two times the same - driver stopped. it is not possible to change ( there it is the battery code )
tried to change the capacity on daly bms app ( one from 280ah to 279ah ) ->same problem.
daly gives back the battery-code (my cas 20220119 on both daly's ) - so it is not empty an the capacity is not added to string
maybe the capacity could be added allways to the identifier ?

back to 2023_05_31 workes fine.
if you want, we can try together

**Reply by mr-manuel** [reply]
`20220119` should be a custom string you can change with the PC software.

**Reply by andzie82** [reply]
Does this mean that it’s now possible to connect several packs with individual JK-BMS and they will show up in GX as individual cells and the GX will monitor them individually?

**Reply by Approved-OK** [reply]
Took a while for editing the Battery-Code but it works now. thx Louis

@andzie82 : i only user Daly BMS with cti-RS485 converter vom Modbus rtu . each of the two batterypacks has it's own daly bms which is connectet do cerbo usb

![2024-04-28 08_15_44VRM Portal](https://github.com/Louisvdw/dbus-serialbattery/assets/130371031/e2a7cb2a-f549-41d7-8371-4ac2daca4510)

To enable view, use remote-console:   Einstellungen - SystemSetup - Batteriemessungen

![image](https://github.com/Louisvdw/dbus-serialbattery/assets/130371031/906bfba6-4da6-494d-8c63-e92c18ba201f)

**Reply by mr-manuel** [reply]
I‘m not Louis, but you‘re welcome :-)

**Comment by mr-manuel** [comment]
Could you please test this? https://github.com/Louisvdw/dbus-serialbattery/issues/1022#issuecomment-2104675732

**Comment by kpenkov98** [comment]
Not sure if this is still relevant but I've got 3 Daly BMS connected as parallel to a raspi with your driver (stable version) It sees all 3 but because of the identifier the other 2 batteries dont show up. Serial connection is used, im going to try and change the serial number on them but changing battery capacity did not work.
